src/ErrorProne.NET/ErrorProne.NET.Test/ExceptionHandlingRules/ArgumentExceptionUsesStringAnalyzerTests.cs

```diff
@@ -0,0 +1,154 @@
+using ErrorProne.NET.Rules.ExceptionHandling;
+using NUnit.Framework;
+using RoslynNunitTestRunner;
+
+namespace ErrorProne.NET.Test.ExceptionHandling
+{
+    [TestFixture]
+    public class ArgumentExceptionUsesStringAnalyzerTests : CSharpAnalyzerTestFixture<ArgumentExceptionUsesStringAnalyzer> {
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasNoParametersInCtor() {
+            var code = @"
+using System;
+class Test
+{
+    public Test(string arg) 
+    {
+        throw [|new ArgumentException()|];
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionParamNameRequired);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasNoParametersInMethod() {
+            var code = @"
+using System;
+class Test
+{
+    public Foo(string arg) 
+    {
+        throw [|new ArgumentException()|];
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionParamNameRequired);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasOnlyMessageParameterInCtor() {
+            var code = @"
+using System;
+class Test
+{
+    public Test(string arg) 
+    {
+        throw [|new ArgumentException(""It's an argument exception"")|];
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionParamNameRequired);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasOnlyMessageParameterInMethod() {
+            var code = @"
+using System;
+class Test
+{
+    public Foo(string arg) 
+    {
+        throw [|new ArgumentException(""It's an argument exception"")|];
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionParamNameRequired);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasWrongParameterNameInCtor() {
+            var code = @"
+using System;
+class Test
+{
+    public Test(string arg) 
+    {
+        throw new ArgumentException(""It's an argument exception"", [|""arg1""|]);
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionMethodHasNoSuchParamName);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasWrongParameterNameInMethod() {
+            var code = @"
+using System;
+class Test
+{
+    public Foo(string arg) 
+    {
+        throw new ArgumentException(""It's an argument exception"", [|""arg1""|]);
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionMethodHasNoSuchParamName);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasStringParameterNameInCtor() {
+            var code = @"
+using System;
+class Test
+{
+    public Test(string arg) 
+    {
+        throw new ArgumentException(""It's an argument exception"", [|""arg""|]);
+    }
+}";
+            HasDiagnostic(code, RuleIds.ArgumentExceptionParamNameShouldNotBeString);
+        }
+
+        [Test]
+        public void ShouldWarnOnArgumentExceptionHasStringParameterNameInMethod() {
```

Comment:
I think explicit test case and even special message would be highly useful (but maybe not part of this PR) when user uses different order of arguments - `throw ArgumentExcpetion("name", "message")`.